proxy_test.go

```diff
@@ -768,6 +768,43 @@ func TestShimSendStdinAfterExeccmd(t *testing.T) {
 	rig.Stop()
 }
 
+// This test ensures that a shim will receive the exit code signal even if
+// the process inside the corresponding container has not been started.
+// This case occurs when a container has been created and it receives a KILL
+// or TERM signal. We don't forward the signal to hyperstart because nothing
+// runs inside the VM, but we still need to remove the shim which is seen as
+// the container process from any orchestrator.
+func testShimSignalProcessNotStarted(t *testing.T, signal syscall.Signal) {
+	rig := newTestRig(t)
+	rig.Start()
+
+	token := rig.RegisterVM()
+	shim := rig.ServeNewShim(token)
+	session := peekIOSession(rig.proxy, token)
+
+	session.containerID = testContainerID
+
+	// Don't close session.processStarted because we want to simulate
+	// a case where the container has been created but not started.
+
+	// Send KILL signal and check the shim receives an exit code.
+	err := shim.client.Kill(signal)
+	assert.Nil(t, err)
```

Comment:
I still think it would be good to test with another signal (not `SIGTERM` or `SIGKILL`).

Couldn't you change this function to be `testShimSignalProcessNotStarted(..., expectFail bool)` and then assert that sending `SIGUSR1` does fail here?